docs/docs/accessibility.md

```diff
@@ -0,0 +1,320 @@
+Although these standard HTML practices can be directly introduced into React, note that the `for` attribute is written as `htmlFor` in JSX:
+
+```javascript{1}
+<label htmlFor="namedInput">Name:</label>
+<input id="namedInput" type="text" name="name"/>
+```
+
+### Notifying the user of errors
+
+Error situations need to be understood by all users. The following link show us how to expose error texts to screen readers as well:
+
+- [The W3C demonstrates user notifications](https://www.w3.org/WAI/tutorials/forms/notifications/)
+- [WebAIM looks at form validation](http://webaim.org/techniques/formvalidation/)
+
+## Focus control
+
+Ensure that your web application can be fully operated with the keyboard only: 
+
+- [WebAIM talks about keyboard accessibility](https://webaim.org/techniques/keyboard/)
+
+### Keyboard focus and focus outline
+
+Keyboard focus refers to the current element in the DOM that is selected to accept input from the keyboard. We see it everywhere as a focus outline similar to the that shown on the following image:
+
+<center><img src="/react/img/docs/keyboard-focus.png" alt="Blue keyboard focus outline around a selected link." /></center>
+
+Only ever use CSS that removes this outline, for example by setting `outline:0`, if you are replacing it with another focus outline implementation.
+
+### Mechanisms to skip to desired content
+
+Provide a mechanism to allow users to skip past content sections in your application as this assists and speeds up keyboard navigation.
+
+#### Semantic HTML and screen readers
+
+Use landmark elements and roles, such as `<main>` and `<aside>`, to demarcate page regions as some assistive technologies allow the user to quickly
+navigate to these regions.
+
+Read more about the use of these elements to enhance accessibility here:
+
+- [Deque University - HTML 5 and ARIA Landmarks](https://dequeuniversity.com/assets/html/jquery-summit/html5/slides/landmarks.html)
+
+#### Skiplinks
+
+Skiplinks or Skip Navigation Links are hidden navigation links that only become visible when keyboard users interact with the page. They are very easy to implement with 
+internal page anchors and some styling:
+
+- [WebAIM - Skip Navigation Links](http://webaim.org/techniques/skipnav/)
+
+### Programmatically managing focus
+
+Our React applications continuously modify the HTML DOM during runtime, sometimes leading to keyboard focus being lost or set to an unexpected element. In order to repair this, 
+we need to programmatically nudge the keyboard focus in the right direction. For example by resetting keyboard focus to a button that opened a modal window after it was closed.
+
+The Mozilla Developer Network takes a look at this and describes how we can build [keyboard-navigable JavaScript widgets](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets).
+
+To set focus in React, we can use [Refs to Components](refs-and-the-dom.html) or [ReactDOM](react-dom.html).
+
+Using refs:
```

Comment:
Originally I had a different example but I ended up using this one as this exact example exists on the `refs` page in the docs. Perhaps it should be looked at, at some stage then.

I want to try and propose a compromise. And know when I started up with React this was something that took a while to sink in. So I have cut down the code to just show how you focus, instead of showing where you put the code. I still think that maybe it could help to get devs started quicker with focus management.

What do you think of the new code examples? If you still feel edgy about it I will remove it :-)

Changed in next push.